pkg/ovs/ovsctl/ofctl.go

```diff
@@ -87,6 +87,37 @@ func (c *ovsCtlClient) DumpGroups(args ...string) ([][]string, error) {
 	return groupList, nil
 }
 
+func (c *ovsCtlClient) DumpPortsDesc() ([][]string, error) {
+	portsDescDump, err := c.RunOfctlCmd("dump-ports-desc")
+	if err != nil {
+		return nil, err
+	}
+	portsDescStr := strings.TrimSpace(string(portsDescDump))
+	scanner := bufio.NewScanner(strings.NewReader(portsDescStr))
+	scanner.Split(bufio.ScanLines)
+	// Skip the first line.
+	scanner.Scan()
+
+	rawPortDescItems := make([][]string, 0)
+	var portItem []string
+	for scanner.Scan() {
+		str := scanner.Text()
+		// If the line starts with a port number, it should be the first line of an OF port. There should be some
+		// subsequent lines to describe the status of the current port, which start with multiple while-spaces.
+		if string(str[1]) != " " {
```

Comment:
I suggest checking the length first to avoid panic.
Although there may be impossible to have a zero length string, the check is cheap and may prevent some future changes.

Reply:
Sure, I will add check for the length of `str`